testing/integration-tests/src/full_client/client/mod.rs

```diff
@@ -442,227 +253,28 @@ async fn unsigned_extrinsic_is_same_shape_as_polkadotjs() {
 }
 
 #[tokio::test]
-async fn rpc_state_call() -> Result<(), subxt::Error> {
-    let ctx = test_context().await;
-    let api = ctx.client();
-
-    // get metadata via state_call.
-    let (_, meta1) = api
-        .rpc()
-        .state_call::<(Compact<u32>, Metadata)>("Metadata_metadata", None, None)
-        .await?;
-
-    // get metadata via `state_getMetadata`.
-    let meta2 = api.rpc().metadata_legacy(None).await?;
-
-    // They should be the same.
-    assert_eq!(meta1.encode(), meta2.encode());
-
-    Ok(())
-}
-
-#[tokio::test]
-async fn chainhead_unstable_follow() {
-    let ctx = test_context().await;
-    let api = ctx.client();
-
-    // Check subscription with runtime updates set on false.
-    let mut blocks = api.rpc().chainhead_unstable_follow(false).await.unwrap();
```

Comment:
I think this is fine since we plan to add them back under a dedicated backend for the Spec V2.

In the meanwhile, I remember this is used by [polakadot-interceptor here](https://github.com/paritytech/polkadot-introspector/blob/7284fc638d241d23063f4a1607fb158c76a856ff/essentials/src/api/subxt_wrapper.rs#L693-L701).  However I think, the same could be achieved with the new Subxt API (for reference @AndreiEres ) which will allow seamless transition between `chainHead` and the legacy RPC.

IIRC this was used to create a metric about the duration time between two reported blocks (either best / finalized), we could work together here to find an appropriate solution with the new API (we might merge the 2 streams for best and finalized)

Reply:
Thank you for mentioning me! I should've expected it by `unstable` mark in `chainHead` call :-) 

So I see that I should switch our subscription in introspector to new API, right? Which one, btw?

Reply:
After we release the next subxt version, merging the [Backend::stream_best_block_headers](https://github.com/paritytech/subxt/pull/1126/files#diff-17bc33d8a1d22f8fcb082c5eff935dfa21b562c9d9133b520c357e9975e48bdaR170) and [Backend::stream_finalized_block_headers](https://github.com/paritytech/subxt/pull/1126/files#diff-17bc33d8a1d22f8fcb082c5eff935dfa21b562c9d9133b520c357e9975e48bdaR183) should produce the same information as the `chainHead`

Reply:
Would this call also work (if all you care about is the time between _any_ new block)?

```rust
api.blocks().subscribe_all().await
```

or 

```rust
api.blocks().subscribe_best().await
```

If you care about all new "best" blocks.

Reply:
Thank you